src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -13,6 +13,25 @@ public static float l2Squared(float[] queryVector, float[] inputVector) {
         return squaredDistance;
     }
 
+    public static float cosinesimilOptimized(float[] queryVector, float[] inputVector, double normQueryVector) {
+        double dotProduct = 0.0f;
+        double normInputVector = 0.0f;
+        if (normQueryVector == -1) {
```

Comment:
this check should preferably be avoided for every input vector

Reply:
can this check be moved to be done exactly once at the factory when the script is created?

Reply:
Removed this check as this case does not occur